src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -142,7 +142,7 @@ IWeakReference* GetComWeakReference(OBJECTREF* pObject)
     if (pMT->IsComObjectType()
      && (pMT == g_pBaseCOMObject || !pMT->IsExtensibleRCW()))
     {
-        pWeakReferenceSourceUnsafe = reinterpret_cast<IWeakReferenceSource*>(GetComIPFromObjectRef(pObject, IID_IWeakReferenceSource, false /* throwIfNoComIP */));
+        pWeakReferenceSource = reinterpret_cast<IWeakReferenceSource*>(GetComIPFromObjectRef(pObject, IID_IWeakReferenceSource, false /* throwIfNoComIP */));
```

Comment:
Do me a favor and please double check that assignment for a `SafeComHolder<T>` doesn't do an `AddRef()`. For `ATL::CComPtr<T>` I believe it does and so I would normally called `ATL::CComPtr<T>::Attach()` instead, but don't know the rules for `SafeComHolder<T>`.

Reply:
I've validated that `SafeComHolder<T>` does not `AddRef()` when assigned a new value.